components/password_reset_form/password_reset_form.tsx

```diff
@@ -52,7 +53,11 @@ export default class PasswordResetForm extends React.PureComponent {
         this.setState({error: null});
 
         const token = (new URLSearchParams(this.props.location.search)).get('token');
-        const {data, error} = await this.props.actions.resetUserPassword(token, password);
+        if (typeof token === null) {
+            throw new Error('token must be a string');
+        }
+        const safeToken = token as string;
+        const {data, error} = await this.props.actions.resetUserPassword(safeToken, password);
```

Comment:
there are several things here I would do differently

1. `typeof variable === null` is not valid, since the type for `null` is Object. Also typeof always returns a string and cannot return `null`, so this would have never been resulted in `true`.
2. It typically is not necessary to cast a type on a return value (in this case `const safeToken = token as string;`), since the return type is `string` (technically it is `USVString`, but since the type definition for it was `type USVString = string;` the developers deemed it useless)
3. there is a unnecessary duplication by re-assigning the `token` to `safeToken`

My suggestion here would be to first discriminate by using a different if-condition and then just calling the action with the token

```suggestion
        if (typeof token !== 'string') {
            throw new Error('token must be a string');
        }

        const {data, error} = await this.props.actions.resetUserPassword(token, password);
```

Reply:
Done